src/LodeRunner.API.Test/UnitTests/IntervalCheckerTest.cs

```diff
@@ -0,0 +1,155 @@
+            {
+                logger.Setup(logLevelConfig: config, logValues: config, projectName: projectName);
+            });
+
+            return loggerFactory.CreateLogger<IntervalCheckerTest>();
+        }
+
+        /// <summary>
+        /// Calculate the maximum waiting time.
+        /// </summary>
+        /// <param name="interval">The interval.</param>
+        /// <returns>the max waiting time.</returns>
+        private static int CalculateTimeout(int interval)
+        {
+            int newInterval = interval;
+
+            for (int i = 1; i < 3; i++)
+            {
+                newInterval /= 2;
+                interval += newInterval;
+            }
+
+            // We will wait for Max Interval time plus 1 extra second.
+            return interval + 1;
+        }
+
+        /// <summary>
+        /// Runs the interval checker and waits until timeout and get cancellation token status and console output.
+        /// </summary>
+        /// <param name="taskToExecute">The task to execute.</param>
+        /// <param name="intervalSeconds">The interval seconds.</param>
+        /// <param name="retryLimit">The retry limit.</param>
+        /// <returns>The cancellationToken status and the outputAsStringList.</returns>
+        private async Task<bool> RunIntervalCheckerWaitAndGetCancellationStatus(Func<Task<bool>> taskToExecute, int intervalSeconds = 4, int retryLimit = 3)
+        {
+            CancellationTokenSource cancellationTokenSource = new ();
+
+            using var intervalChecker = new IntervalChecker(taskToExecute, this.logger, cancellationTokenSource, intervalSeconds, retryLimit);
+
+            intervalChecker.Start();
+
+            return await this.WaitForTimeoutAndGetCancellationRequestStatus(cancellationTokenSource, intervalSeconds);
+        }
+
+        /// <summary>
+        /// Waits until timeout is reached and determines if Cancellation request is in place.
+        /// </summary>
+        /// <param name="cancellationTokenSource">The cancellation token source.</param>
+        /// <param name="intervalSeconds">The interval seconds.</param>
+        /// <returns>Whether the cancellation was requested or not.</returns>
+        private async Task<bool> WaitForTimeoutAndGetCancellationRequestStatus(CancellationTokenSource cancellationTokenSource, int intervalSeconds)
+        {
+            int maxWaitingTime = CalculateTimeout(intervalSeconds);
+            await Task.Delay(maxWaitingTime * 1000).ConfigureAwait(false);
+
+            bool cancellationRequested = cancellationTokenSource.IsCancellationRequested;
+            this.output.WriteLine($"Total Wait time: {maxWaitingTime} secs.\tCancellation Requested: {cancellationRequested}\t");
+            return cancellationRequested;
+        }
+
+        /// <summary>
+        /// Gets a boolean true.
+        /// </summary>
+        /// <returns>true.</returns>
+        private Task<bool> GetBooleanTrue()
```

Comment:
1. Seems like you could use one method to return True or False.  Probably make it return any scalar value by declaring it correctly as a generic.  
2. Use a name that is more meaningful.  So, maybe instead of `GetBooleanFalse` use a name like `ReturnScalarValueAsTask` if it is changed to support any scalar or `ReturnBoolValueAsTask` if it stays specific.

Reply:
Refactored as Generic